sopel/bot.py

```diff
@@ -648,3 +646,71 @@ def cap_req(self, module_name, capability, arg=None, failure_callback=None,
             entry.append(_CapReq(prefix, module_name, failure_callback, arg,
                                  success_callback))
             self._cap_reqs[cap] = entry
+
+    def register_url_callback(self, pattern, callback):
+        """Register a ``callback`` for URLs matching the regex ``pattern``
+
+        :param pattern: compiled regex pattern to register
+        :param callback: callable object to handle matching URLs
+
+        .. versionadded: 7.0.0
+
+            This method has been added in Sopel 7.0.0 in order to offer
+            a unified interface for URL callbacks.
+
+        """
+        if not self.memory.contains('url_callbacks'):
+            self.memory['url_callbacks'] = tools.SopelMemory()
+        self.memory['url_callbacks'][pattern] = callback
+
+    def unregister_url_callback(self, pattern):
+        """Unregister the callback for URLs matching the regex ``pattern``
+
+        :param pattern: compiled regex pattern to unregister callback
+
+        .. versionadded: 7.0.0
+
+            This method has been added in Sopel 7.0.0 in order to offer
+            a unified interface to URL callbacks.
```

Comment:
Similar to the `register_` function docstring, it would be useful here to show module authors what to replace in their `shutdown` methods.

Reply:
Done.